cirq-ft/cirq_ft/algos/and_gate.py

```diff
@@ -134,15 +135,18 @@ def _decompose_via_tree(
             )
 
     def decompose_from_registers(
-        self, *, context: cirq.DecompositionContext, **quregs: Sequence[cirq.Qid]
+        self,
+        *,
+        context: cirq.DecompositionContext,
+        **quregs: NDArray[cirq.Qid],  # type:ignore[type-var]
     ) -> cirq.OP_TREE:
         control, ancilla, target = quregs['control'], quregs['ancilla'], quregs['target']
         if len(self.cv) == 2:
             yield self._decompose_single_and(
                 self.cv[0], self.cv[1], control[0], control[1], *target
             )
         else:
-            yield self._decompose_via_tree(control, self.cv, ancilla, *target)
+            yield self._decompose_via_tree(control.tolist(), self.cv, ancilla.tolist(), *target)
```

Comment:
why do we need `tolist()`? Do we need to update the type signatures for `_decompose_via_tree` if we're using list-specific things

Reply:
We need this mainly because `Sequence[cirq.Qid]` is not compatible with `NDArray[cirq.Qid]`.

I'll update the signature of `_decompose_via_tree` to accept `NDArray[cirq.Qid]` instead.

Reply:
that's kindof annoying